pysrc/scripts/sampling.py

```diff
@@ -81,6 +99,41 @@ def clean_results(results_folder: str) -> None:
     for tmp_file in result_files:
         os.remove(os.path.join(os.getcwd(), results_folder, tmp_file))
 
+def get_file_combinations(samples: list[tuple[str, str, float]], config: GlobalConfiguration) \
+        -> list[tuple[str, str, float]]:
+    data_type_dict: dict[int, list[str]] = {}
+    for num_files_index in range(0, len(samples)):
+        for sample_file_index in range(0, len(samples[num_files_index])):
+
+            current_tuple = samples[num_files_index][sample_file_index]
+            path_to_data = current_tuple[0]
```

Comment:
This currently doesn't make sense. `samples` is `list[tuple[str, str, float]]`. Therefore, `samples[num_file_index]` is `tuple[str, str, float]`. You're now indexing into this `tuple` which is still (technically) fine, but accessing `[0]` on that value is not what you want to do.